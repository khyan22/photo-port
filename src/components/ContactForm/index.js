import React, { useState } from 'react';

function ContactForm() {
  const [ formState, setFormState ] = useState({
    name: '',
    email: '',
    message: ''
  });

  const { name, email, message } = formState;

  function handleChange(event) {
      setFormState({ ...formState, [event.target.name]: event.target.value });
  };

  function handleSubmit(event) {
    event.preventDefault();
    console.log(formState);
  };

  return(
    <section>
      <h1>Contact Form</h1>
      <form id='contact-form' onSubmit={handleSubmit}>
        <div>
          <label htmlFor='name'>Name:</label>
          <input type='text' name='name' defaultValue={name} onChange={handleChange} />
        </div>
        <div>
          <label htmlFor='email'>Email:</label>
          <input type='email' name='email' defaultValue={email} onChange={handleChange}/>
        </div>
        <div>
          <label htmlFor='message'>Message:</label>
          <textarea type='text' name='message' defaultValue={message} rows='5' onChange={handleChange} />
        </div>
        <button type='submit'>Submit</button>
      </form>
    </section>
  );
};

export default ContactForm;