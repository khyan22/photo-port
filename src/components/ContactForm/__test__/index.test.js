import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactForm from '..';

afterEach(cleanup);

describe('ContactForm component', () => {
  it('renders', () => {
    render(<ContactForm/>);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ContactForm/>);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('text renders', () => {
  it('renders h1 text', () => {
    const { getByTestId } = render(<ContactForm/>);

    expect(getByTestId('h1tag')).toHaveTextContent('Contact Me');
  });

  it('renders btn text', () => {
    const { getByTestId } = render(<ContactForm/>);

    expect(getByTestId('submit')).toHaveTextContent('Submit');
  });
});
