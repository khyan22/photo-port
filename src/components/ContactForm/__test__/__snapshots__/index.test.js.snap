// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Contact Me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
          type="text"
        />
      </div>
      <button
        data-testid="submit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
